src/ImageMetadata.jl

```diff
@@ -41,7 +41,9 @@ mutable struct ImageMeta{T,N,A<:AbstractArray,P<:AbstractDict{String,Any}} <: Ab
     end
 end
 ImageMeta(data::AbstractArray{T,N}, props::AbstractDict{String,Any}) where {T,N} = ImageMeta{T,N,typeof(data),typeof(props)}(data,props)
+ImageMeta(data::AbstractArray, props::AbstractDict{<:AbstractString}) = ImageMeta(data, convert(Dict{String,Any}, props))
```

Comment:
I suspect we should insist on `props::Dict{<:AbstractString}` here, rather than generalizing it to `AbstractDict`. If someone passes a `SortedDict`, this will un-sort it. I suspect if they went to the trouble to construct it as a `SortedDict` they'd be unhappy about that.

Reply:
If it isn't AbstractDict the constructor becomes recursive. Would you prefer that line 43 becomes
``` ImageMeta(data::AbstractArray{T,N}, props::Dict{String,Any}) where {T,N} = ImageMeta{T,N,typeof(data),typeof(props)}(data,props)```

Reply:
Oh, you're right. You could add another "non-copying" method
```julia
ImageMeta(data::AbstractArray{T,N}, props::Dict{String,Any}) where {T,N} = ImageMeta{T,N,typeof(data),typeof(props)}(data,props)
```
That will be more specific than the `Dict{<:AbstractString}` method.

Reply:
Is there anyway to get from a ```TypeName``` to the type itself? We could use ```typeof(props).name``` to get the right type name to keep the other types of Dicts working automagically.

Reply:
Not generically. I could create a type like this:

```
DictWithNElements{N,K,V} <: AbstractDict{K,V}
```

What do you do with that `N` parameter in front of the others?

Reply:
I had this in mind. Maybe just too much. Nothing I need now.
```ImageMeta(data::AbstractArray, props::AbstractDict{<:AbstractString}) = ImageMeta(data, convert(type_from_typename(typeof(props).name){String,Any}, props))```

Reply:
Goodness! I'm sorry about this. I've been working on a custom dictionary for image metadata [here](https://github.com/Tokazama/ImageFormats.jl/blob/master/src/imageproperties.jl) that may be helpful. Unfortunately I've been spread pretty thin with my projects lately so it's not yet a finished work.

Reply:
Thanks for the help, learned some stuff.

Reply:
This is nice work, particularly for a first PR. Much appreciated.